### ISSUE TITLE
[BUG] MKS3+ stuck at sd card read (probably - see description)

### ISSUE THREAD

**Issue by arekm** [prusa3d/Prusa-Firmware#3568]
**Printer type** - MK3S+
**Printer firmware version** -  own build of MK3 branch as of 3ebd8b88a4d1f6879979d7afddd67d430416e6d8

**SD card or USB/Octoprint**
Printing from SD card.

**Describe the bug**

Printer was printing for few hours and then it stopped. Pause & resume didn't help (procedure finished
but printer was back in a stuck state).

Note - I'm using sdwire hardware for sdcard which could be causing trouble, so I'm not sure if this is really a printer firmware bug. sdwire is basically a switch which can disconnect sdcard lines from the printer (but it shouldn't do that unless told to do so which wasn't the case here).

So far happened two times with different micro sd cards and different sd card adapters. And I'm using sdwire for a weeks already. The second time it was stuck just at the end of intro line (note: mine intro line is a bit longer than default).

Questions:
- printer was not reporting any sd reading errors, nothing on LCD, nothing on serial. But does it report sd card problems if these occur or just silently tries to deal with these?

- is it possible to pause a print, remove sd card and insert it again (or maybe even new card with the same file) and resume a print ? (didn't try that but I'm wondering for next time when this problem happens)

Dump and firmware hex attached.

```
2022-08-20 19:32:14,649 - Send: M27
2022-08-20 19:32:16,138 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:16,144 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:16,145 - Recv: 03:42
2022-08-20 19:31:27,102 - Recv: T:251.5 /250.0 B:89.7 /90.0 T0:251.5 /250.0 @:38 B@:105 P:0.0 A:44.4
2022-08-20 19:31:27,341 - Send: M27
2022-08-20 19:31:27,660 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:27,664 - Recv: SD printing byte 3964777/8303993
2022-08-20 19:31:27,666 - Recv: 03:41
2022-08-20 19:31:27,668 - Recv: ok
2022-08-20 19:31:28,344 - Send: M27
2022-08-20 19:31:28,512 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:28,514 - Recv: SD printing byte 3965526/8303993
2022-08-20 19:31:28,517 - Recv: 03:41
2022-08-20 19:31:28,518 - Recv: ok
2022-08-20 19:31:29,109 - Recv: T:250.2 /250.0 B:89.7 /90.0 T0:250.2 /250.0 @:52 B@:90 P:0.0 A:44.3
2022-08-20 19:31:29,347 - Send: M27
2022-08-20 19:31:29,826 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:29,830 - Recv: SD printing byte 3966495/8303993
2022-08-20 19:31:29,832 - Recv: 03:41
2022-08-20 19:31:29,833 - Recv: ok
2022-08-20 19:31:30,223 - Recv: NORMAL MODE: Percent done: 50; print time remaining in mins: 216; Change in mins: -1
2022-08-20 19:31:30,229 - Send: M114
2022-08-20 19:31:30,231 - Recv: SILENT MODE: Percent done: 50; print time remaining in mins: 222; Change in mins: -1
2022-08-20 19:31:30,535 - Recv: X:136.50 Y:154.47 Z:19.10 E:278.18 Count X: 156.82 Y:155.02 Z:19.41 E:276.66
2022-08-20 19:31:30,544 - Recv: ok
2022-08-20 19:31:30,546 - Send: M27
2022-08-20 19:31:30,683 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:30,684 - Recv: SD printing byte 3967195/8303993
2022-08-20 19:31:30,687 - Recv: 03:41
2022-08-20 19:31:30,688 - Recv: ok
2022-08-20 19:31:31,112 - Recv: T:253.2 /250.0 B:89.7 /90.0 T0:253.2 /250.0 @:16 B@:76 P:0.0 A:44.6
2022-08-20 19:31:31,353 - Send: M27
2022-08-20 19:31:31,678 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:31,682 - Recv: SD printing byte 3967834/8303993
2022-08-20 19:31:31,683 - Recv: 03:41
2022-08-20 19:31:31,684 - Recv: ok
2022-08-20 19:31:32,357 - Send: M27
2022-08-20 19:31:32,505 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:32,506 - Recv: SD printing byte 3968471/8303993
2022-08-20 19:31:32,509 - Recv: 03:41
2022-08-20 19:31:32,510 - Recv: ok
2022-08-20 19:31:33,119 - Recv: T:247.4 /250.0 B:90.4 /90.0 T0:247.4 /250.0 @:80 B@:0 P:0.0 A:44.5
2022-08-20 19:31:33,360 - Send: M27
2022-08-20 19:31:33,599 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:33,603 - Recv: SD printing byte 3969386/8303993
2022-08-20 19:31:33,604 - Recv: 03:41
2022-08-20 19:31:33,605 - Recv: ok
2022-08-20 19:31:34,363 - Send: M27
2022-08-20 19:31:35,127 - Recv: T:245.7 /250.0 B:89.7 /90.0 T0:245.7 /250.0 @:100 B@:57 P:0.0 A:44.7
2022-08-20 19:31:37,134 - Recv: T:247.8 /250.0 B:89.9 /90.0 T0:247.8 /250.0 @:78 B@:30 P:0.0 A:44.8
2022-08-20 19:31:38,793 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:38,794 - Recv: SD printing byte 3970318/8303993
2022-08-20 19:31:38,795 - Recv: 03:41
2022-08-20 19:31:38,800 - Recv: ok
2022-08-20 19:31:38,802 - Send: M27
2022-08-20 19:31:39,141 - Recv: T:250.5 /250.0 B:90.0 /90.0 T0:250.5 /250.0 @:48 B@:22 P:0.0 A:44.6
2022-08-20 19:31:41,148 - Recv: T:250.1 /250.0 B:90.1 /90.0 T0:250.1 /250.0 @:54 B@:11 P:0.0 A:44.5
2022-08-20 19:31:41,369 - Recv: echo:busy: processing
2022-08-20 19:31:43,154 - Recv: T:249.1 /250.0 B:90.0 /90.0 T0:249.1 /250.0 @:65 B@:31 P:0.0 A:44.3
2022-08-20 19:31:43,729 - Recv: NORMAL MODE: Percent done: 50; print time remaining in mins: 215; Change in mins: -1
2022-08-20 19:31:43,738 - Recv: SILENT MODE: Percent done: 50; print time remaining in mins: 222; Change in mins: -1
2022-08-20 19:31:45,159 - Recv: T:252.2 /250.0 B:90.3 /90.0 T0:252.2 /250.0 @:30 B@:0 P:0.0 A:44.5
2022-08-20 19:31:45,748 - Recv: echo:busy: processing
2022-08-20 19:31:47,165 - Recv: T:251.6 /250.0 B:90.1 /90.0 T0:251.6 /250.0 @:37 B@:34 P:0.0 A:44.6
2022-08-20 19:31:47,748 - Recv: echo:busy: processing
2022-08-20 19:31:49,177 - Recv: T:248.8 /250.0 B:90.0 /90.0 T0:248.8 /250.0 @:69 B@:63 P:0.0 A:44.3
2022-08-20 19:31:51,643 - Recv: T:250.0 /250.0 B:89.9 /90.0 T0:250.0 /250.0 @:55 B@:80 P:0.0 A:44.6
2022-08-20 19:31:54,656 - Communication timeout while printing, trying to trigger response from printer. Configure long running commands or increase communication timeout if that happens regularly on specific commands or long moves.
2022-08-20 19:31:54,667 - Send: M27
2022-08-20 19:31:54,690 - Recv: T:249.3 /250.0 B:90.6 /90.0 T0:249.3 /250.0 @:63 B@:0 P:0.0 A:44.4
2022-08-20 19:31:54,695 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:31:54,698 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:31:54,699 - Recv: 03:41
2022-08-20 19:31:54,700 - Recv: ok
2022-08-20 19:31:54,701 - Send: M114
2022-08-20 19:31:56,693 - Recv: T:250.2 /250.0 B:89.7 /90.0 T0:250.2 /250.0 @:53 B@:115 P:0.0 A:45.3
2022-08-20 19:31:56,703 - Recv: echo:busy: processing
2022-08-20 19:31:59,528 - Recv: T:249.6 /250.0 B:89.7 /90.0 T0:249.6 /250.0 @:60 B@:89 P:0.0 A:45.3
2022-08-20 19:32:02,535 - Recv: T:249.3 /250.0 B:89.6 /90.0 T0:249.3 /250.0 @:64 B@:108 P:0.0 A:44.5
2022-08-20 19:32:05,544 - Communication timeout while printing, trying to trigger response from printer. Configure long running commands or increase communication timeout if that happens regularly on specific commands or long moves.
2022-08-20 19:32:05,578 - Recv: T:249.7 /250.0 B:89.6 /90.0 T0:249.7 /250.0 @:59 B@:108 P:0.0 A:44.6
2022-08-20 19:32:08,585 - Communication timeout while printing, trying to trigger response from printer. Configure long running commands or increase communication timeout if that happens regularly on specific commands or long moves.
2022-08-20 19:32:08,591 - Recv: T:250.6 /250.0 B:89.6 /90.0 T0:250.6 /250.0 @:48 B@:120 P:0.0 A:44.6
2022-08-20 19:32:10,092 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:10,097 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:10,099 - Recv: 03:42
2022-08-20 19:32:10,100 - Recv: ok
2022-08-20 19:32:10,103 - Send: M105
2022-08-20 19:32:11,604 - Recv: T:251.6 /250.0 B:89.8 /90.0 T0:251.6 /250.0 @:38 B@:73 P:0.0 A:44.7
2022-08-20 19:32:11,616 - Recv: X:26.06 Y:109.69 Z:19.10 E:349.20 Count X: 26.01 Y:108.86 Z:19.29 E:349.20
2022-08-20 19:32:11,618 - Recv: ok
2022-08-20 19:32:11,627 - Send: M27
2022-08-20 19:32:13,119 - Recv: ok T:252.9 /250.0 B:90.1 /90.0 T0:252.9 /250.0 @:22 B@:2 P:0.0 A:45.2
2022-08-20 19:32:13,128 - Send: M27
2022-08-20 19:32:13,130 - Recv: ok
2022-08-20 19:32:13,490 - Send: M27
2022-08-20 19:32:14,632 - Recv: T:252.7 /250.0 B:91.0 /90.0 T0:252.7 /250.0 @:24 B@:0 P:0.0 A:44.7
2022-08-20 19:32:14,644 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:14,645 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:14,646 - Recv: 03:42
2022-08-20 19:32:14,647 - Recv: ok
2022-08-20 19:32:14,649 - Send: M27
2022-08-20 19:32:16,138 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:16,144 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:16,145 - Recv: 03:42
2022-08-20 19:32:16,146 - Recv: ok
2022-08-20 19:32:16,148 - Send: M27
2022-08-20 19:32:17,650 - Recv: T:250.4 /250.0 B:90.5 /90.0 T0:250.4 /250.0 @:50 B@:0 P:0.0 A:44.7
2022-08-20 19:32:17,660 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:17,661 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:17,662 - Recv: 03:42
2022-08-20 19:32:17,663 - Recv: ok
2022-08-20 19:32:17,665 - Send: M27
2022-08-20 19:32:19,165 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:19,170 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:19,172 - Recv: 03:42
2022-08-20 19:32:19,172 - Recv: ok
2022-08-20 19:32:19,175 - Send: M27
2022-08-20 19:32:20,677 - Recv: T:249.9 /250.0 B:90.7 /90.0 T0:249.9 /250.0 @:56 B@:0 P:0.0 A:44.6
2022-08-20 19:32:20,686 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:20,688 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:20,689 - Recv: 03:42
2022-08-20 19:32:20,690 - Recv: ok
2022-08-20 19:32:20,692 - Send: M27
2022-08-20 19:32:22,217 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:22,224 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:22,227 - Recv: 03:42
2022-08-20 19:32:22,229 - Recv: ok
2022-08-20 19:32:22,232 - Send: M27
2022-08-20 19:32:23,728 - Recv: T:250.4 /250.0 B:90.8 /90.0 T0:250.4 /250.0 @:48 B@:0 P:0.0 A:44.9
2022-08-20 19:32:23,736 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:23,739 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:23,740 - Recv: 03:42
2022-08-20 19:32:23,741 - Recv: ok
2022-08-20 19:32:23,742 - Send: M27
2022-08-20 19:32:25,243 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:25,249 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:25,250 - Recv: 03:42
2022-08-20 19:32:25,251 - Recv: ok
2022-08-20 19:32:25,253 - Send: M27
2022-08-20 19:32:26,751 - Recv: T:253.4 /250.0 B:90.8 /90.0 T0:253.4 /250.0 @:14 B@:0 P:0.0 A:44.5
2022-08-20 19:32:26,759 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:26,762 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:26,763 - Recv: 03:42
2022-08-20 19:32:26,764 - Recv: ok
2022-08-20 19:32:26,765 - Send: M27
2022-08-20 19:32:28,267 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:28,269 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:28,278 - Recv: 03:42
2022-08-20 19:32:28,279 - Recv: ok
2022-08-20 19:32:28,283 - Send: M27
2022-08-20 19:32:29,774 - Recv: T:253.4 /250.0 B:90.8 /90.0 T0:253.4 /250.0 @:13 B@:0 P:0.0 A:44.7
2022-08-20 19:32:29,785 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:29,787 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:29,788 - Recv: 03:42
2022-08-20 19:32:29,789 - Recv: ok
2022-08-20 19:32:29,791 - Send: M27
2022-08-20 19:32:31,290 - Recv: fixed-length-upper-bridge_0.6n_0.3mm_PETG_MK3S_7h14m
2022-08-20 19:32:31,295 - Recv: SD printing byte 3970560/8303993
2022-08-20 19:32:31,296 - Recv: 03:42
2022-08-20 19:32:31,297 - Recv: ok
```

I was using own firmware build from MK3 branch with latest 3ebd8b88a4d1f6879979d7afddd67d430416e6d8 commit.
[Firmware-intl.hex.txt](https://github.com/prusa3d/Prusa-Firmware/files/9389609/Firmware-intl.hex.txt)

Dump, as seen on video I've tried to do dump. That's what was stored:
[dump.txt](https://github.com/prusa3d/Prusa-Firmware/files/9389620/dump.txt)

**Video**
https://www.youtube.com/watch?v=qWh7y4qZMLA&feature=youtu.be
Tried to pause and resume at beginning.
Notice how slowly it reacts to knob.
Also after I stopped end went into "Print from SD" menu it showed only one entry "Main" - huh? It should be "Back" or some contents.

**Comment by arekm** [comment]
Maybe the problem is with sd card slot?

https://www.printables.com/model/15388-prusa-mk3-sd-card-slot-fix-not-reading-the-card-at.

Answer for my first question in initial issue comment would shed some light here...

(note, problem didn't repeat so far, yet)

**Comment by Prusa-Support** [comment]
Hello! Thank you for your message, sorry for the late response. Normally, if card was removed (or not reading correctly, "as if it was removed"), there should be a message indicating this (https://cdn.help.prusa3d.com/wp-content/uploads/cardremoved-600x320.jpg). I tested the pausing-removing card-reinserting, it seemed that resuming the print still worked regularly though I wouldn't be able to guarantee it fully for all cases. 

Will note this for our devs, in case of more info from our side, we'll expand on it on this thread. 

Alessandro Pantaleo
Prusa Research
